src/coreclr/nativeaot/Runtime/windows/PalRedhawkMinWin.cpp

```diff
@@ -137,6 +145,8 @@ void InitializeCurrentProcessCpuCount()
 // initialization and false on failure.
 REDHAWK_PALEXPORT bool REDHAWK_PALAPI PalInit()
 {
+    g_mainThreadId = GetCurrentThreadId();
```

Comment:
The initialization can run on any thread when NativeAOT is used to produce libraries.

Reply:
That is inconvenient. 

There is no easy way to identify the main thread. 
Online samples suggest iterating through all threads and look for the earliest start time.

Reply:
A part of the problem is that the thread object in NativeAOT lives on the stack and thus is owned by the thread, so we must unlink the threads from the list and must take a lock.